Sources/AWSLambdaRuntimeCore/Lambda+LocalServer.swift

```diff
@@ -46,62 +45,269 @@ extension Lambda {
 
 // MARK: - Local Mock Server
 
-private enum LocalLambda {
-    struct Server {
-        private let logger: Logger
-        private let group: EventLoopGroup
-        private let host: String
-        private let port: Int
-        private let invocationEndpoint: String
+public protocol LocalLambdaInvocationProxy {
+    init(eventLoop: EventLoop)
+
+    // TODO: in most (all?) cases we do not need async interface
+
+    /// throws `LocalLambda.InvocationError` if request is invalid
+    func invocation(from request: LocalLambda.HTTPRequest) -> EventLoopFuture<ByteBuffer>
+    func processResult(_ result: ByteBuffer?) -> EventLoopFuture<LocalLambda.HTTPResponse>
+    func processError(_ error: ByteBuffer?) -> EventLoopFuture<LocalLambda.HTTPResponse>
+}
```

Comment:
I think in most cases there is no need for async calls.

In a typical scenario a HTTP request would be used to create new HTTP request with different payload.

Async interface would be useful in a fairly exotic scenario (as far as local testing/development is concerned); for example, if the `LocalLambdaInvocationProxy` was connecting with external server to set valid Authorization header.